packages/api/src/models/sensorModel.js

```diff
@@ -61,10 +61,20 @@ class Sensor extends Model {
     };
   }
 
-  static async createSensor(sensor, farm_id, user_id, partnerId) {
+  static async createSensor(sensor, farm_id, user_id, partner_id) {
     const trx = await transaction.start(Model.knex());
 
     try {
+      const existingSensor = await LocationModel.getSensorLocation(
```

Comment:
@anant00729 I did it multiple times inside the map function because the logic to return the correct success and error sensors with the proper data depends on the `sensorLocations` and `registeredSensors` arrays being the same length and order. If we change this, then we have to re-write a very large portion of the logic for the route.

Reply:
okay got it.